dsgrid/registry/dataset_registry_manager.py

```diff
@@ -127,37 +143,123 @@ def _check_lookup_data_consistency(self, config: DatasetConfig, load_data_lookup
                     f"load_data_lookup records do not match dimension records for {name}"
                 )
 
-    def _check_dataset_time_consistency(self, config: DatasetConfig, load_data):
-        time_dim = config.get_dimension(DimensionType.TIME)
-        tz = time_dim.get_tzinfo()
-        time_ranges = time_dim.get_time_ranges()
-        assert len(time_ranges) == 1, len(time_ranges)
-        time_range = time_ranges[0]
-        # TODO: need to support validation of multiple time ranges: DSGRID-173
-
-        expected_timestamps = time_dim.list_time_range(time_range)
-        actual_timestamps = [
-            x.timestamp.astimezone().astimezone(tz)
-            for x in load_data.select("timestamp").distinct().sort("timestamp").collect()
-        ]
-        if expected_timestamps != actual_timestamps:
-            mismatch = sorted(
-                set(expected_timestamps).symmetric_difference(set(actual_timestamps))
+    def _check_one_table_data_consistency(self, config: DatasetConfig, load_data_one_table):
+        """
+        Check all data dimensions, including Metric records.
+        For Time, only check that time_col_name = timestamp
+        """
+        dimension_types = []
+        metric_cols = []
+
+        for col in load_data_one_table.columns:
+            if col.lower() == "time":
```

Comment:
It may be better to raise an exception if `timestamp` is not a column.

Reply:
but growth rates would not have a timestamp col.

Reply:
Ok. I think my main point is that it seems odd to check only for the invalid column "time". Other invalid names are not checked. Can we check explicitly for all required columns?

Reply:
And can we reuse a lot of the validations already created for the two-table format?